File: CommonLanguageExtensions_vb/TimeSpanExtensions.vb
Imports System.Runtime.CompilerServices

Namespace CommonLanguageExtensions
    ''' <summary>
    ''' TimeSpan extensions
    ''' </summary>
    Public Module TimeSpanExtensions
        ''' <summary>
        ''' Format a TimeSpan with AM PM
        ''' </summary>
        ''' <param name="sender">TimeSpan to format</param>
        ''' <param name="format">Optional format</param>
        ''' <returns></returns>
        <Extension>
        Public Function Formatted(sender As TimeSpan, Optional format As String = "hh:mm tt") As String
            Return Date.Today.Add(sender).ToString(format)
        End Function
    End Module
End Namespace

File: CommonLanguageExensions/TimeSpanExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CommonLanguageExtensions
{
    public static class TimeSpanExtensions
    {
        /// <summary>
        /// Format a TimeSpan with AM PM
        /// </summary>
        /// <param name="sender">TimeSpan to format</param>
        /// <param name="format">Optional format</param>
        /// <returns></returns>
        public static string Formatted(this TimeSpan sender, string format = "hh:mm tt") => DateTime.Today.Add(sender).ToString(format);
    }
}
